services/api/pepc_boost_test.go

```diff
@@ -113,19 +117,180 @@ func assertTobTxs(t *testing.T, backend *testBackend, slot uint64, parentHash st
+	require.NoError(t, err)
+
+	invalidWethDaiTxContents := common.LoadFileContents(t, "../../testdata/traces/custom/invalid_weth_dai_tx.json")
+	invalidWethDaiTx := new(gethtypes.Transaction)
+	err = invalidWethDaiTx.UnmarshalJSON(invalidWethDaiTxContents)
+	require.NoError(t, err)
+
+	validWethDaiTxTraceContents := common.LoadFileContents(t, "../../testdata/traces/custom/valid_weth_dai_tx_trace.json")
+	validWethDaiTxTrace := new(common.CallTrace)
+	err = json.Unmarshal(validWethDaiTxTraceContents, validWethDaiTxTrace)
+	require.NoError(t, err)
+
+	invalidWethDaiTraceContents := common.LoadFileContents(t, "../../testdata/traces/custom/invalid_weth_dai_tx_trace.json")
+	invalidWethDaiTrace := new(common.CallTrace)
+	err = json.Unmarshal(invalidWethDaiTraceContents, invalidWethDaiTrace)
+	require.NoError(t, err)
+
+	return validWethDaiTx, validWethDaiTxTrace, invalidWethDaiTx, invalidWethDaiTrace
+}
+
+func GetTestPayloadAttributes(t *testing.T) (string, types.Address, []byte, string, phase0.BLSPubKey, uint64) {
+	t.Helper()
+	parentHash := "0xbd3291854dc822b7ec585925cda0e18f06af28fa2886e15f52d52dd4b6f94ed6"
+	feeRec, err := types.HexToAddress("0x5cc0dde14e7256340cc820415a6022a7d1c93a35")
+	require.NoError(t, err)
+	withdrawalsRoot, err := hexutil.Decode("0xb15ed76298ff84a586b1d875df08b6676c98dfe9c7cd73fab88450348d8e70c8")
+	require.NoError(t, err)
+	prevRandao := "0x9962816e9d0a39fd4c80935338a741dc916d1545694e41eb5a505e1a3098f9e4"
+	proposerPubkeyByte, err := hexutil.Decode(testProposerKey)
+	require.NoError(t, err)
+	proposerPubkey := phase0.BLSPubKey(proposerPubkeyByte)
+
+	return parentHash, feeRec, withdrawalsRoot, prevRandao, proposerPubkey, uint64(32)
+}
+
+func TestIsTxWEthDaiSwap(t *testing.T) {
+	_, _, backend := startTestBackend(t)
+
+	validWethDaiTx, validWethDaiTxTrace, invalidWethDaiTx, invalidWethDaiTrace := GetCustomDevnetTracingRelatedTestData(t)
+
+	cases := []struct {
+		description   string
+		callTraces    *common.CallTrace
+		tx            *gethtypes.Transaction
+		isTxCorrect   bool
+		network       string
+		requiredError string
+	}{
+		{
+			description:   "valid tx",
+			callTraces:    validWethDaiTxTrace,
+			tx:            validWethDaiTx,
+			isTxCorrect:   true,
+			network:       "custom",
+			requiredError: "",
+		},
+		{
+			description:   "invalid tx",
+			callTraces:    invalidWethDaiTrace,
+			tx:            invalidWethDaiTx,
+			isTxCorrect:   false,
+			network:       "custom",
+			requiredError: "",
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.description, func(t *testing.T) {
+			// Payload attributes
+			parentHash, feeRec, withdrawalsRoot, prevRandao, proposerPubkey, headSlot := GetTestPayloadAttributes(t)
+
+			prepareBackend(t, backend, headSlot, parentHash, feeRec, withdrawalsRoot, prevRandao, proposerPubkey, c.network)
+
+			res, err := backend.relay.IsTxWEthDaiSwap(c.callTraces)
+			if c.requiredError != "" {
+				require.Contains(t, err.Error(), c.requiredError)
```

Comment:
can return in if and then move next to outer scope

Reply:
yeah yeah! will refactor all at once!